### PR TITLE
Feat channel settings

### PR DESCRIPTION
**Pull request by giachuvan**

### CODE REVIEW

File: crates/mezon-ui/src/chat/channel_settings/overview_tab.rs [hunk 1]
```diff
@@ -0,0 +1,877 @@
...
+                    this.detail_loading = false;
+                    cx.notify();
+                });
+                return;
+            };
+            let result = api.list_channel_detail(channel_id.get()).await;
+            let _ = this.update(cx, |this, cx| {
+                this.detail_loading = false;
+                if let Ok(detail) = result {
+                    this.saved_topic = detail.topic.clone();
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/overview_tab.rs]: saved_topic is overwritten before is_topic_dirty is read, so the check compares the value against itself. When Channel.topic is empty (ChannelCreated sets it to String::new()) the fetched topic never reaches the input, the tab opens dirty, and Save writes the empty topic back. Capture both dirty flags before assigning. Same for age_restricted below.

File: crates/mezon-ui/src/sidebar/channel_sidebar/menu.rs [hunk 2]
```diff
@@ -158,7 +158,18 @@ fn mark_channel_read(
     }
 }
 
-fn open_channel_settings(
+pub(super) fn can_open_channel_settings(clan_id: ClanId, cx: &App) -> bool {
+    let Some(permissions) = PermissionStore::try_global(cx) else {
+        return false;
+    };
+    let permissions = permissions.read(cx);
+    permissions.check(clan_id, None, PERMISSION_CLAN_OWNER, cx)
```
Comment (huylexuan44) [inline crates/mezon-ui/src/sidebar/channel_sidebar/menu.rs:150]: These constants are no longer imported in menu.rs on develop, so this fails to compile after merge. Use mezon_store::can_manage_channel(clan_id, cx) — check() already returns true for clan owner and covers administrator.

File: crates/mezon-ui/src/chat/channel_settings/overview_tab.rs [hunk 3]
```diff
@@ -0,0 +1,877 @@
...
+        let task = channel_list.update(cx, |store, cx| {
+            store.update_channel_overview(clan_id, channel_id, label, topic, age_restricted, cx)
+        });
+        self._save_task = Some(cx.spawn(async move |this, cx| {
+            let result = task.await;
+            let _ = this.update(cx, |this, cx| {
+                this.saving = false;
+                match result {
+                    Ok(()) => {
+                        this.saved_label = this.draft_label(cx);
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/overview_tab.rs]: On success this stores the current draft, not what was sent. Text typed during the request is marked saved, and since the store trims the label the save bar never clears for a name with surrounding whitespace. Capture label, topic and age_restricted in the closure.

File: crates/mezon-ui/src/chat/channel_settings/mod.rs [hunk 4]
```diff
@@ -355,6 +409,37 @@ impl ChannelSettingScreen {
...
+            div()
+                .w(px(SIDEBAR_ITEM_WIDTH))
+                .mt(px(16.0))
+                .mb(px(16.0))
+                .border_t_1()
+                .border_color(theme.border),
+        );
+        nav = nav.child(
+            div()
+                .id("channel-settings-delete")
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/mod.rs:448]: Delete Channel has no on_click and no hover background. React opens a confirm modal here. Wire it or drop the entry.

File: crates/mezon-ui/src/components/compositions/channel_row_element.rs [hunk 5]
```diff
@@ -53,6 +56,24 @@ pub struct ThreadConnector {
     pub color: Hsla,
 }
 
+pub struct ChannelRowTrailingAction {
+    pub icon: IconName,
+    pub color: Hsla,
```
Comment (huylexuan44) [inline crates/mezon-ui/src/components/compositions/channel_row_element.rs]: color is never read — the gear only paints while hovered and always uses hover_color. Paint the resting state with it or remove the field.

File: crates/mezon-ui/src/chat/channel_settings/mod.rs [hunk 6]
```diff
@@ -267,18 +283,56 @@ impl ChannelSettingScreen {
...
+            }
+            ChannelSettingsTab::Category => {
+                if self.category_tab.is_some() {
+                    return;
+                }
+                let clan_id = self.clan_id;
+                let channel_id = self.channel_id;
+                let settings = self.settings.clone();
+                let tab = cx.new(|cx| CategoryTab::new(clan_id, channel_id, settings, cx));
+                cx.observe(&tab, |_, _, cx| cx.notify()).detach();
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/mod.rs]: detach() leaks one subscription per tab switch; these tabs are dropped and rebuilt each time. Store the Subscription next to the tab entity. Same applies to the Permissions arm above.

File: crates/mezon-store/src/channel.rs [hunk 7]
```diff
@@ -3445,6 +3636,29 @@ fn remove_channel(categories: &mut [Category], channel_id: ChannelId) -> bool {
...
+        .cloned()
+    else {
+        return false;
+    };
+    if !remove_channel(categories, channel_id) {
+        return false;
+    }
+    channel.category_id = Some(new_category_id.to_string());
+    channel.category_name = new_category_name.to_string();
+    insert_channel(categories, channel)
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: If insert_channel fails after remove_channel succeeded, the channel is lost from the cache and the caller stores the mutated list. Insert before removing, or restore on failure.

**Comment by huylexuan44** [comment]
Needs changes before merge. Details are in the inline comments.

- Does not compile against current develop — Lint, Test and Windows Check all fail, and these are this PR's errors, not the pre-existing clippy noise. Two causes: menu.rs uses permission constants develop no longer imports there, and develop added ApiChannelDesc / Channel test constructors after the fork that do not set the new topic, age_restricted, e2ee and app_id fields (merged channel.rs:7990, 8054, 8110). Rebase on develop.
- Data loss: the detail fetch in overview_tab.rs overwrites saved_topic and saved_age_restricted before reading the dirty flags, so the tab can open already dirty and Save writes an empty topic back. sync_from_store gets the order right; the fetch path does not.
- Also worth fixing: save() records the current draft instead of the sent values, Delete Channel is a dead label, ChannelRowTrailingAction.color is unused, the two new tabs detach their observe subscriptions, and channel detail is fetched twice (view on open, store again on save) while the view reads DTO fields directly.
- Good parts: UpdateChannelDescParams over more positional args, the new store tests, and sidebar styling that tracks channelSettingItem.tsx closely. Please add a PR description for a change this size.

File: crates/mezon-store/src/channel.rs [hunk 8]
```diff
@@ -2209,9 +2383,18 @@ impl ChannelList {
             RealtimeEvent::ChannelUpdated(e) => {
                 let id = ChannelId(e.channel_id);
                 let label = (!e.channel_label.is_empty()).then_some(e.channel_label.clone());
+                let topic = Some(e.topic.clone());
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: ChannelUpdated often omits topic/age_restricted; unconditional Some() wipes them. Guard like channel_label above.

File: crates/mezon-store/src/channel.rs [hunk 9]
```diff
@@ -227,6 +229,15 @@ pub enum CreateChannelError {
...
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum UpdateChannelOverviewError {
+    InvalidName,
+    DuplicateName,
+    Other(String),
+}
+
+pub const MAX_CHANNEL_TOPIC_CHARS: usize = 512;
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: React caps topic at 1024, not 512. Existing longer topics get truncated on open and raise a phantom unsaved-changes bar.

File: crates/mezon-client/src/transport.rs [hunk 10]
```diff
@@ -727,12 +727,30 @@ pub struct ApiChannelDesc {
...
+    #[serde(default)]
+    pub app_id: i64,
 }
 
 fn default_channel_active() -> i32 {
     1
 }
 
+#[derive(Debug, Clone, Default)]
+pub struct UpdateChannelDescParams {
```
Comment (huylexuan44) [inline crates/mezon-client/src/transport.rs:759]: app_id missing here, so UpdateChannelDesc always sends 0. React forwards it; App channels can lose their app binding.

File: crates/mezon-store/src/channel.rs [hunk 11]
```diff
@@ -1962,6 +1973,167 @@ impl ChannelList {
...
+                        .await
+                        .map_err(|e| UpdateChannelOverviewError::Other(e.to_string()))?
+                };
+                if duplicate {
+                    return Err(UpdateChannelOverviewError::DuplicateName);
+                }
+            }
+
+            let detail = api
+                .list_channel_detail(channel_id.get())
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: Redundant round-trip: OverviewTab already fetched list_channel_detail on open. Reuse it or put e2ee on Channel.

File: crates/mezon-store/src/channel.rs [hunk 12]
```diff
@@ -3445,6 +3636,33 @@ fn remove_channel(categories: &mut [Category], channel_id: ChannelId) -> bool {
...
+    if !remove_channel(categories, channel_id) {
+        return false;
+    }
+    let mut moved = channel.clone();
+    moved.category_id = Some(new_category_id.to_string());
+    moved.category_name = new_category_name.to_string();
+    if insert_channel(categories, moved) {
+        return true;
+    }
+    insert_channel(categories, channel)
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: Restore path returns true, so the caller treats a failed move as a move. If both inserts fail the channel is dropped.

File: crates/mezon-ui/src/chat/channel_settings/overview_tab.rs [hunk 13]
```diff
@@ -0,0 +1,889 @@
...
+                                        }),
+                                ),
+                        ),
+                ),
+        )
+}
+
+impl Render for OverviewTab {
+    fn render(&mut self, window: &mut Window, cx: &mut Context<Self>) -> impl IntoElement {
+        self.sync_from_store(window, cx);
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/overview_tab.rs]: sync_from_store mutates input state inside render and can emit Change -> notify -> render. Drive it from a subscription.

File: crates/mezon-ui/src/chat/channel_settings/overview_tab.rs [hunk 14]
```diff
@@ -0,0 +1,889 @@
...
+            if *event == TextAreaEvent::Change {
+                this.on_topic_changed(cx);
+            }
+        });
+
+        let channel_list = ChannelList::global(cx);
+        let permission_store = PermissionStore::global(cx);
+        let subs = vec![
+            cx.observe(&settings, |_, _, cx| cx.notify()),
+            cx.observe(&channel_list, |_, _, cx| cx.notify()),
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/overview_tab.rs]: ChannelList notifies on every badge/unread change app-wide, re-rendering this tab. Scope this to the edited channel.

File: crates/mezon-ui/src/sidebar/channel_sidebar/mod.rs [hunk 15]
```diff
@@ -1671,6 +1671,11 @@ fn render_sidebar_item(
             let clan_id_inner = active_clan_id_for_nav;
             let menu_channel_id: ChannelId = ch_id.parse().unwrap_or_default();
             let menu_clan_id: ClanId = clan_id_inner.unwrap_or_default();
+            let show_settings_gear = !*is_thread
```
Comment (huylexuan44) [inline crates/mezon-ui/src/sidebar/channel_sidebar/mod.rs]: can_open_channel_settings runs 4 permission checks per visible row per frame. It only depends on clan_id - hoist it.

File: crates/mezon-ui/src/chat/channel_settings/mod.rs [hunk 16]
```diff
@@ -168,9 +184,19 @@ impl ChannelSettingScreen {
     }
 
     pub fn release_active_tab(&mut self, cx: &mut Context<Self>) {
-        if self.permissions_tab.take().is_some() || !self.clan_id.is_zero() {
+        if self.permissions_tab.take().is_some()
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/mod.rs]: || short-circuits, so only the first Some tab is taken. Take all six into locals, then notify if any was set.

File: crates/mezon-ui/src/chat/channel_settings/overview_tab.rs [hunk 17]
```diff
@@ -0,0 +1,889 @@
...
+                    .h(px(40.))
+                    .flex()
+                    .items_center()
+                    .pl(px(12.))
+                    .pr(px(12.))
+                    .rounded(px(8.))
+                    .border_1()
+                    .border_color(theme.tokens.border_theme_primary)
+                    .bg(theme.tokens.bg_input_secondary)
+                    .when(!can_edit, |el| el.opacity(0.6))
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/overview_tab.rs]: Opacity only - InputState has no disabled state, so read-only users can still type and the text sticks. Disable it.

File: crates/mezon-ui/src/chat/channel_settings/overview_tab.rs [hunk 18]
```diff
@@ -0,0 +1,889 @@
...
+
+    fn is_age_restricted_dirty(&self, _cx: &App) -> bool {
+        self.draft_age_restricted != self.saved_age_restricted
+    }
+
+    fn is_dirty(&self, cx: &App) -> bool {
+        self.is_name_dirty(cx) || self.is_topic_dirty(cx) || self.is_age_restricted_dirty(cx)
+    }
+
+    pub fn should_show_save_bar(&self, cx: &App) -> bool {
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/overview_tab.rs:290]: React hides the save bar while the name is invalid or duplicate. Include the validation state in this condition.

File: crates/mezon-ui/src/chat/channel_settings/category_tab.rs [hunk 19]
```diff
@@ -0,0 +1,283 @@
...
+            .cursor_pointer()
+            .child(
+                div()
+                    .min_w_0()
+                    .flex_1()
+                    .truncate()
+                    .child(category_name.to_uppercase()),
+            )
+            .child(
+                Icon::new(IconName::ArrowDown)
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/category_tab.rs]: React CategoryChannel uses ArrowDownFill, same as the hide-inactivity dropdown. Use IconName::ArrowDownFill.

File: crates/mezon-ui/src/components/compositions/channel_row_element.rs [hunk 20]
```diff
@@ -171,6 +197,23 @@ fn muted_color(mut color: Hsla, muted: bool) -> Hsla {
...
+    let row_height = row_bounds.size.height;
+    let gear_x = row_bounds.origin.x + row_bounds.size.width - GEAR_RIGHT_GAP - GEAR_SIZE;
+    let gear_y = row_bounds.origin.y + (row_height - GEAR_SIZE) / 2.;
+    Bounds {
+        origin: point(gear_x, gear_y),
+        size: size(GEAR_SIZE, GEAR_SIZE),
+    }
+}
+
+fn contains_point(bounds: Bounds<Pixels>, point: Point<Pixels>) -> bool {
```
Comment (huylexuan44) [inline crates/mezon-ui/src/components/compositions/channel_row_element.rs]: gpui Bounds::contains already does this. Drop contains_point.

File: crates/mezon-ui/src/components/compositions/channel_row_element.rs [hunk 21]
```diff
@@ -53,6 +56,22 @@ pub struct ThreadConnector {
     pub color: Hsla,
 }
 
+pub struct ChannelRowTrailingAction {
+    pub icon: IconName,
+    pub hover_color: Hsla,
+    pub on_click: ClickHandler,
+}
+
+impl Clone for ChannelRowTrailingAction {
```
Comment (huylexuan44) [inline crates/mezon-ui/src/components/compositions/channel_row_element.rs]: ClickHandler is an Rc, so derive(Clone) works. Drop the manual impl.

File: crates/mezon-ui/src/chat/channel_settings/overview_tab.rs [hunk 22]
```diff
@@ -0,0 +1,889 @@
...
+
+use crate::app::shell::Shell;
+use crate::components::primitives::{
+    Button, ButtonVariants, Icon, IconName, Input, InputEvent, InputState, Switch, TextArea,
+    TextAreaEvent, TextAreaField, h_flex, v_flex,
+};
+use crate::theme::{ActiveTheme, Theme};
+use crate::util::assets::{CHANNEL_SETTING_LOGO_DARK, CHANNEL_SETTING_LOGO_LIGHT};
+
+fn theme_is_light(theme: &Theme) -> bool {
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/overview_tab.rs]: Duplicate of theme_is_light in chat/stream.rs. Extract one shared helper.

**Comment by huylexuan44** [comment]
CI is red on Lint, Test and Windows Check. The head commit compiles on its own, but the merge with develop does not - the branch is based on an older develop.

- develop narrowed the mezon_store import block in channel_sidebar/menu.rs, so the merged file loses PERMISSION_CLAN_OWNER, PERMISSION_ADMINISTRATOR and PERMISSION_MANAGE_CLAN that can_open_channel_settings needs (3x E0425).
- develop added a favorite_ids parameter to apply_clan_structure and removed favorite_extras; ~15 mezon-store test call sites still use the old signature.
- Some test fixtures were updated for the new Channel.topic/age_restricted and ApiChannelDesc.topic/age_restricted/e2ee/app_id fields, others were not (channel.rs 8256, 8320, 8376).

Merge develop into the branch and fix the fixtures, then re-run. Inline comments cover the functional findings.
